contrib/ipp/src/pjhufftbl.c

```diff
@@ -64,21 +64,21 @@ const int mfxown_pj_extend_test[16] =
 const int mfxown_pj_extend_offset[16] =
 {
   0,
-  ((-1) <<  1) + 1,
-  ((-1) <<  2) + 1,
-  ((-1) <<  3) + 1,
-  ((-1) <<  4) + 1,
-  ((-1) <<  5) + 1,
-  ((-1) <<  6) + 1,
-  ((-1) <<  7) + 1,
-  ((-1) <<  8) + 1,
-  ((-1) <<  9) + 1,
-  ((-1) << 10) + 1,
-  ((-1) << 11) + 1,
-  ((-1) << 12) + 1,
-  ((-1) << 13) + 1,
-  ((-1) << 14) + 1,
-  ((-1) << 15) + 1
+  ((-1u) <<  1) + 1,
```

Comment:
I believe this will produce some warnings on other compilers. Maybe we can use something different like:
```((~0u) << 1) + 1,```
@lakulako what do you think?

Reply:
Regardless of the selected approach, I would suggest to introduce some macro like;
`#define MINUS_ONE (-1u)`
and use it instead. I have some feeling that satisfying all the compilers here might be challenging and the above will be easier to differentiate.

Reply:
done